### PR TITLE
Tweak: Added responsive capability to Icon Position control in Icon Box widget (#3040) [OBXT-573]

### PR DESCRIPTION
**Pull request by hein-obox**

### CODE REVIEW

File: includes/widgets/icon-box.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 use Elementor\Core\Kits\Documents\Tabs\Global_Colors;
 use Elementor\Core\Kits\Documents\Tabs\Global_Typography;
+use Elementor\Core\Breakpoints\Manager as Breakpoints_Manager;
```
Comment (udidol) [inline includes/widgets/icon-box.php:10]: You didn't actually use it...

File: assets/dev/scss/frontend/widgets/icon-list.scss [hunk 2]
```diff
@@ -86,7 +86,8 @@
 		.elementor-icon-list-icon {
 			text-align: var(--e-icon-list-icon-align);
 
-			svg {
+			svg,
+			i {
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-list.scss:90]: ?

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 3]
```diff
@@ -1,48 +1,90 @@
...
 
-				.elementor-icon-box-img {
-					margin: auto;
-				}
+			.elementor-icon-box-icon {
+				margin-bottom: var(--icon-box-icon-margin, 15px);
+				margin-right: auto;
+				margin-left: auto;
 			}
+
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:69]: no need for this space

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 4]
```diff
@@ -1,48 +1,90 @@
...
+				text-align: $start;
+				flex-direction: getValueByDirection(row, row-reverse);
 			}
 
-			&left {
+			.elementor-icon-box-icon {
+				margin-right: var(--icon-box-icon-margin);
+				margin-left: 0;
+				margin-bottom: unset;
+			}
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:49]: no need for this space

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 5]
```diff
@@ -1,48 +1,90 @@
...
+				text-align: $end;
+				flex-direction: getValueByDirection(row-reverse, row);
 			}
 
-			&right {
+			.elementor-icon-box-icon {
+				margin-left: var(--icon-box-icon-margin);
+				margin-right: 0;
+				margin-bottom: unset;
+			}
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:34]: no need for this space

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 6]
```diff
@@ -1,48 +1,90 @@
...
+				display: flex;
+			}
 
-				.elementor-icon-box-wrapper {
-					display: flex;
-				}
+			.elementor-icon-box-icon {
+				display: inline-flex;
+				flex: 0 0 auto; // Hack for Safari
+			}
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:19]: no need for this space

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 7]
```diff
@@ -1,48 +1,90 @@
...
+
+			.elementor-icon-box-wrapper {
+				text-align: $end;
+				flex-direction: getValueByDirection(row-reverse, row);
 			}
 
-			&right {
+			.elementor-icon-box-icon {
+				margin-left: var(--icon-box-icon-margin);
+				margin-right: 0;
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:31]: Please test this right/left thing in RTL mode to make sure it works as it should in both LTR and RTL

File: assets/dev/scss/frontend/widgets/icon-box.scss [hunk 8]
```diff
@@ -1,48 +1,90 @@
...
+		&top {
+
+			.elementor-icon-box-wrapper {
+				display: block;
+				text-align: center;
+				flex-direction: unset;
 			}
 
-			&top {
+			.elementor-icon-box-img {
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-box.scss:60]: What is this class?

**Review by udidol — CHANGES_REQUESTED** [review]

File: includes/widgets/icon-box.php [hunk 9]
```diff
@@ -178,12 +179,17 @@ protected function register_controls() {
 			]
 		);
 
-		$this->add_control(
+		$position_align_args = [
+			'alignment_control_prefix_class' => 'elementor%s-position-',
+		];
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply includes/widgets/icon-box.php:184]: I use it for the prefix class, see line 207: 

`'prefix_class' => $position_align_args['alignment_control_prefix_class'],`

File: assets/dev/scss/frontend/widgets/icon-list.scss (hunk 2 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-list.scss:90]: changed

File: assets/dev/scss/frontend/widgets/icon-box.scss (hunk 3 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss:69]: changed

File: assets/dev/scss/frontend/widgets/icon-box.scss (hunk 4 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss:49]: changed

File: assets/dev/scss/frontend/widgets/icon-box.scss (hunk 5 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss:34]: changed

File: assets/dev/scss/frontend/widgets/icon-box.scss (hunk 8 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss:60]: You mean this class ".elementor-icon-box-img"?
I am actually not sure. 
If copied and pasted it from the original version.

File: includes/widgets/icon-box.php (hunk 9 above)
[the remark this answers is not in the document]
Comment (udidol) [reply includes/widgets/icon-box.php:184]: @hein-obox

File: assets/dev/scss/frontend/widgets/icon-box.scss (hunk 7 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss]: I am studying this aspect now

Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-box.scss:30]: @udidol 
Reverted to the original non rtl/ltr friendly version.

File: includes/widgets/icon-box.php [hunk 10]
```diff
@@ -740,4 +737,4 @@ protected function content_template() {
 	public function on_import( $element ) {
 		return Icons_Manager::on_import_migration( $element, 'icon', 'selected_icon', true );
 	}
-}
+}
```
Comment (udidol) [inline includes/widgets/icon-box.php:740]: Add an empty line at the end of this file please

File: core/kits/manager.php [hunk 11]
```diff
@@ -40,9 +40,9 @@ public function get_active_kit() {
 	public function get_active_kit_for_frontend() {
 		$kit = Plugin::$instance->documents->get_doc_for_frontend( $this->get_active_id() );
 
-		if ( ! $this->is_valid_kit( $kit ) ) {
-			return $this->get_empty_kit_instance();
-		}
+		// if ( ! $this->is_valid_kit( $kit ) ) {
+		// 	return $this->get_empty_kit_instance();
+		// }
```
Comment (udidol) [inline core/kits/manager.php:45]: What is this? @hein-obox 
I think you might have gotten mixed up with Louis on this one.

File: core/documents-manager.php [hunk 12]
```diff
@@ -257,11 +300,18 @@ public function get_doc_or_auto_save( $id, $user_id = 0 ) {
...
+		// } else {
+		// 	$document = $this->get( $post_id );
+		// }
+
+		// echo 'mykit'.$post_id.'<br />';
+		// echo get_the_title( $post_id );
+
+		$document = $this->get( $post_id );
+
+		// var_dump( $document );
```
Comment (udidol) [inline core/documents-manager.php:314]: Same here @hein-obox Please talk to Louis, make sure there's no mixed up code.

File: core/documents-manager.php [hunk 13]
```diff
@@ -174,14 +174,20 @@ public function register_document_type( $type, $class ) {
 	 * @return false|Document Document data or false if post ID was not entered.
 	 */
 	public function get( $post_id, $from_cache = true ) {
+
```
Comment (udidol) [inline core/documents-manager.php:177]: Same for this file.

File: assets/dev/scss/frontend/widgets/icon-list.scss [hunk 14]
```diff
@@ -86,7 +86,7 @@
 		.elementor-icon-list-icon {
 			text-align: var(--e-icon-list-icon-align);
 
-			svg {
```
Comment (udidol) [inline assets/dev/scss/frontend/widgets/icon-list.scss:89]: I don't understand what is this `i` element addition

**Review by udidol — CHANGES_REQUESTED** [review]

File: includes/widgets/icon-box.php (hunk 10 above)
Comment (hein-obox) [reply includes/widgets/icon-box.php]: added

File: core/kits/manager.php (hunk 11 above)
Comment (hein-obox) [reply core/kits/manager.php:45]: * Sorry, I accidentely merged some other files while working late *

File: core/documents-manager.php (hunk 12 above)
Comment (hein-obox) [reply core/documents-manager.php:314]: * Sorry, I accidentely merged some other files while working late *

File: core/documents-manager.php (hunk 13 above)
Comment (hein-obox) [reply core/documents-manager.php:177]: * Sorry, I accidentely merged some other files while working late *

File: assets/dev/scss/frontend/widgets/icon-list.scss (hunk 14 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/widgets/icon-list.scss:89]: * Fixed *

**Review by udidol — APPROVED** [review]

**Pull request merged**